src/som/interpreter/actors/EventualMessage.java

```diff
@@ -203,10 +215,10 @@ public boolean getHaltOnPromiseMessageResolution() {
    * after the promise is resolved.
    */
   public abstract static class AbstractPromiseSendMessage extends PromiseMessage {
-    private final SSymbol    selector;
-    protected Actor          target;
-    protected Actor          finalSender;
-    protected final SPromise originalTarget;
+    private final SSymbol selector;
+    protected Actor       target;
+    protected Actor       finalSender;
+    protected SPromise    originalTarget;
```

Comment:
Could you at least make it `@CompilationFinal` then? This is likely a performance regression.